src/Test/StateMachine/Parallel.hs

```diff
@@ -498,170 +502,68 @@ shrinkAndValidateNParallel sm = \shouldShrink  (ParallelCommands prefix suffixes
-                     => StateMachine model cmd m resp
-                     -> NParallelCommands cmd resp
-                     -> PropertyM m [(History cmd resp, model Concrete, Logic)]
-runNParallelCommands = runNParallelCommandsNTimes 10
-
-runNParallelCommandsWithSetup :: (Show (cmd Concrete), Show (resp Concrete))
-                     => (Rank2.Traversable cmd, Rank2.Foldable resp)
-                     => (MonadMask m, MonadUnliftIO m)
-                     => m (StateMachine model cmd m resp)
-                     -> NParallelCommands cmd resp
-                     -> PropertyM m [(History cmd resp, model Concrete, Logic)]
-runNParallelCommandsWithSetup = runNParallelCommandsNTimesWithSetup 10
-
-runParallelCommandsNTimes :: (Show (cmd Concrete), Show (resp Concrete))
                           => (Rank2.Traversable cmd, Rank2.Foldable resp)
                           => (MonadMask m, MonadUnliftIO m)
-                          => Int -- ^ How many times to execute the parallel program.
-                          -> StateMachine model cmd m resp
-                          -> ParallelCommands cmd resp
-                          -> PropertyM m [(History cmd resp, model Concrete, Logic)]
-runParallelCommandsNTimes n sm = runParallelCommandsNTimesWithSetup n (pure sm)
-
-runParallelCommandsNTimesWithSetup :: (Show (cmd Concrete), Show (resp Concrete))
-                          => (Rank2.Traversable cmd, Rank2.Foldable resp)
-                          => (MonadMask m, MonadUnliftIO m)
-                          => Int -- ^ How many times to execute the parallel program.
-                          -> m (StateMachine model cmd m resp)
+                          => StateMachine model cmd m resp
                           -> ParallelCommands cmd resp
-                          -> PropertyM m [(History cmd resp, model Concrete, Logic)]
-runParallelCommandsNTimesWithSetup n msm cmds =
-  replicateM n $ do
+                          -> PropertyM m (History cmd resp, model Concrete, Logic)
+runParallelCommands sm cmds = do
     hchan <- newTChanIO
-    ((hist, model, reason1, reason2), sm') <- run $
+    ((hist, model, reason1, reason2), ()) <- run $
       fst <$> generalBracket
-              msm
-              (\sm' ec -> case ec of
-                            ExitCaseSuccess ((_, model, _, _), _) -> cleanup sm' model
-                            _ -> getChanContents hchan >>= cleanup sm' . mkModel sm' . History
+              (pure ())
+              (\_ ec -> case ec of
+                            ExitCaseSuccess ((_, model, _, _), _) -> cleanup sm model
+                            _ -> getChanContents hchan >>= cleanup sm . mkModel sm . History
               )
-              (\sm' -> (,sm') <$> executeParallelCommands sm' cmds hchan True)
-    return (hist, model, logicReason (combineReasons [reason1, reason2]) .&& linearise sm' hist)
+              (\_ -> (,()) <$> executeParallelCommands sm cmds hchan True)
```

Comment:
Can `(,())` be removed? (If you think of datatypes as algebra then `(x, ())` is equivalent to `x * 1` (`== x`)).

Reply:
I don't think it can, it is just to satisfy the types of `generalBracket`.